vendor/engines/secure_rooms/db/migrate/20170409125103_create_occupancies.rb

```diff
@@ -0,0 +1,27 @@
+class CreateOccupancies < ActiveRecord::Migration
+
+  def change
+    create_table :secure_rooms_occupancies do |t|
+      t.references :product, null: false
+      t.references :user, null: false
+      t.references :account
+
+      t.references :entry_event
+      t.datetime :entry_at
+      t.references :exit_event
+      t.datetime :exit_at
+
+      t.datetime :orphaned_at
+
+      t.timestamps null: false
+    end
+
+    add_foreign_key :secure_rooms_occupancies, :products
```

Comment:
This is an annoyance, but oracle doesn't automatically create indexes on foreign keys, so we'll also need to `add_index`. I forget, but I think you might be able to add `index: true` on the `references` as well.

Also, not a big deal (as long as this migration runs both up and down as is), but I believe you can add `foreign_key: true` to the `references` option and get this for free.